packages/react-components/react-virtualizer/library/src/hooks/useMeasureList.ts

```diff
@@ -58,8 +58,19 @@ export function useMeasureList<
   };
 
   React.useEffect(() => {
-    widthArray.current = new Array(totalLength).fill(defaultItemSize);
-    heightArray.current = new Array(totalLength).fill(defaultItemSize);
+    const newHeightLength = totalLength - heightArray.current.length;
+    const newWidthLength = totalLength - widthArray.current.length;
+    // Ensure we grow or truncate arrays with prior properties
```

Comment:
ToDo: Add comment that while we truncate/extend array, if the items in middle change it will recalc the rendered values and keep whitespace consistent.